### ISSUE TITLE
form_data.dart Case Sensitive Issue

### ISSUE THREAD

**Issue by jonilaw** [flutterchina/dio#958]
### New Issue Checklist

- [v] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   3.0.10   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Hi, I have to manually edit the **form_data.dart** file, in order to work correctly with my web server (the header on top of the file uploaded to web server was not truncated). I wonder if this will help, but if it is OK to use Title case on the header shown below in the future release:

![image](https://user-images.githubusercontent.com/20550525/95779187-cc5a7a80-0cf3-11eb-89b2-9a087ffcd7a0.png)

**Content-Disposition
Content-Type
Content-Transfer-Encoding**

Postman and Insomnia is also using the Title case format.

Thanks.
